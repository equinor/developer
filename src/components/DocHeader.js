import React from "react";
import styled from "styled-components";
import { FullWidth } from "./FullWidth";
import { Tag } from "./Tags";
import { getCircleIcon } from "../util/iconUtil";
import { HeaderTitle } from "./Titles";
import { colors } from "../ui";

export const DocHeader = ({ title, tags }) => {
  const HeaderWrapper = styled.div`
    background-color: #f2f2f2;
    width: 100%;
    text-align: center;
  `;

  const IconWrapper = styled.div`
    height: 72px;
    transform: translate(0, -50%);
  `;
  const iconTag = tags && tags.length > 0 ? tags[0] : "";
  return (
    <FullWidth backgroundColor="#f2f2f2">
      <HeaderWrapper>
        <IconWrapper>{getCircleIcon(iconTag)}</IconWrapper>
        <HeaderTitle
          title={title}
          color={colors.mossGreen}
          alignCenter={true}
        />
        <div style={{ padding: "15px 0 20px", textAlign: "center" }}>
          <Tag tag={iconTag} />
        </div>
      </HeaderWrapper>
    </FullWidth>
  );
};
