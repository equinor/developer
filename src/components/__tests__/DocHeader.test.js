import React from "react";
import { DocHeader } from "../DocHeader";
import renderer from "react-test-renderer";

describe("DocHeader", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<DocHeader tags={["Api"]} title="Documents" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders aligned", () => {
    const tree = renderer
      .create(<DocHeader tags={["Api"]} title="Documents" alignCenter />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
