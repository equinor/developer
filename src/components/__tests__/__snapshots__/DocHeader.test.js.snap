// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocHeader renders aligned 1`] = `
.c0 {
  left: calc((100% - 100vw) / 2);
  position: relative;
  width: 100vw;
  padding-left: calc((100vw - 100%) / 2);
  padding-right: calc((100vw - 100%) / 2);
  background-color: #f2f2f2;
}

.c6 {
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
  cursor: inherit;
  color: #007079;
  text-transform: uppercase;
}

.c4 {
  height: 4px;
  width: 20%;
  margin: 20px auto 0;
  background-color: #007079;
}

.c3 {
  color: #333333;
  font-family: Equinor;
  font-size: 52px;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  line-height: 52px;
  text-align: center;
}

.c1 {
  background-color: #f2f2f2;
  width: 100%;
  text-align: center;
}

.c2 {
  height: 72px;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

.c5 {
  font-size: 25px;
  color: black;
  font-weight: 200;
  padding: 0 2px 0 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <test-file-stub />
    </div>
    <div
      className="c3"
      color="#007079"
      title="Documents"
    >
      Documents
    </div>
    <div
      className="c4"
      color="#007079"
    />
    <div
      style={
        Object {
          "padding": "15px 0 20px",
          "textAlign": "center",
        }
      }
    >
      <span
        className="c5"
      >
        /
      </span>
      <span
        className="c6"
      >
        Api
      </span>
    </div>
  </div>
</div>
`;

exports[`DocHeader renders correctly 1`] = `
.c0 {
  left: calc((100% - 100vw) / 2);
  position: relative;
  width: 100vw;
  padding-left: calc((100vw - 100%) / 2);
  padding-right: calc((100vw - 100%) / 2);
  background-color: #f2f2f2;
}

.c6 {
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
  cursor: inherit;
  color: #007079;
  text-transform: uppercase;
}

.c4 {
  height: 4px;
  width: 20%;
  margin: 20px auto 0;
  background-color: #007079;
}

.c3 {
  color: #333333;
  font-family: Equinor;
  font-size: 52px;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  line-height: 52px;
  text-align: center;
}

.c1 {
  background-color: #f2f2f2;
  width: 100%;
  text-align: center;
}

.c2 {
  height: 72px;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

.c5 {
  font-size: 25px;
  color: black;
  font-weight: 200;
  padding: 0 2px 0 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <test-file-stub />
    </div>
    <div
      className="c3"
      color="#007079"
      title="Documents"
    >
      Documents
    </div>
    <div
      className="c4"
      color="#007079"
    />
    <div
      style={
        Object {
          "padding": "15px 0 20px",
          "textAlign": "center",
        }
      }
    >
      <span
        className="c5"
      >
        /
      </span>
      <span
        className="c6"
      >
        Api
      </span>
    </div>
  </div>
</div>
`;
